### PR TITLE
audit: fixes for test checks 

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

As far as I can tell, the regular expression from https://github.com/Homebrew/brew/pull/1530 never actually worked:

```
(def test|test do).*(#{Regexp.escape(HOMEBREW_PREFIX)}/bin/)?#{cmd}[\(\s]+['"]#{Regexp.escape(name)}[\s'"]
```

Because `cmd` is one of `system shell_output pipe_output`, it would match lines like `/usr/local/bin/system "hello"` but never `system "/usr/local/bin/hello"`.

It would be nice to be able to use `HOMEBREW_PREFIX` in cops but sadly that's not available for some reason. Regardless, this is a improvement over the previously broken behavior. It now also checks for `sbin`.

Since I was here, I also moved https://github.com/Homebrew/brew/pull/4663 into a cop.

### CODE REVIEW

File: Library/Homebrew/rubocops/class_cop.rb [hunk 1]
```diff
@@ -22,6 +22,27 @@ def autocorrect(node)
...
+          test_calls(find_block(body_node, :test)) do |node, params|
+            p1, p2 = params
+            if match = string_content(p1).match(%r{^(/usr/local/(s?bin))})
+              offending_node(p1)
+              problem "use \#{#{match[2]}} instead of #{match[1]} in #{node}"
+            end
+
+            if node == :shell_output && node_equals?(p2, 0)
+              offending_node(p2)
+              problem "Passing 0 to shell_output() is redundant"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/rubocops/class_cop.rb]: Fancy adding an autocorrect for this?

File: Library/Homebrew/rubocops/class_cop.rb [hunk 2]
```diff
@@ -22,6 +22,27 @@ def autocorrect(node)
...
         end
       end
+
+      class TestCalls < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          test_calls(find_block(body_node, :test)) do |node, params|
+            p1, p2 = params
+            if match = string_content(p1).match(%r{^(/usr/local/(s?bin))})
+              offending_node(p1)
+              problem "use \#{#{match[2]}} instead of #{match[1]} in #{node}"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/rubocops/class_cop.rb]: Fancy adding an autocorrect for this?

**Comment by MikeMcQuaid** [comment]
Looks good! Autocorrects would be lovely but not essential.

**Comment by jonchang** [comment]
* added autocorrect and tests for same

**Comment by MikeMcQuaid** [comment]
Nice work! Merge when CI is 🍏

**Comment by jonchang** [comment]
Thanks for the review Mike!

**Comment by DomT4** [comment]
This is breaking when formulae don't have a `test do` block, seemingly. Try `brew audit vrpn`, for example.

**Comment by jonchang** [comment]
https://github.com/Homebrew/brew/pull/4710 should fix it.

**Pull request merged**